docs/content/direct/architecture.md

```diff
@@ -476,8 +476,10 @@ When processing a reference to a `Binding` object that still exists, the transpo
 
 Transport controller is based on the controller design patten and aims to bring the current state to the desired state. If a WEC was removed from the `Binding`, the transport controller will also make sure to remove the matching wrapped object(s) from the WEC's mailbox namespace.
 
-The transport controller maintains a cached set of customization properties for each destination, and an association between `Binding` and the set of destinations that it references. When a relevant informer delivers an event that indicates a change in a destination's properties, the controller invalidates that destination's entry in the cache and enqueues a reference to every relevant `Binding`. The template expansion work described above has a side-effect of putting entries in the destination properties cache.
\ No newline at end of file
+To support efficient application of the `CustomTransform` objects, the transport controller maintains a cache that maps a `metav1.GroupResource` to the result of digesting all the relevant `CustomTransform` objects. The data type `customTransformCollection` holds the cache. This cache is populated as needed in the course of transforming workload objects. To support eventually removing an entry from that cache when no longer needed, each entry also tracks the set of `Binding` objects that depend on that cache entry; when that set becomes empty, the entry is removed from the cache. In order to know when a `Binding` has changed from dependent to not dependent, the cache also tracks the set of `metav1.GroupResource` relevant to each `Binding`. Also: in order to know what to update when the Spec of a `CustomTransform` changes, the cache also tracks the spec of each `CustomTransform` that contributed to an entry in the cache. When such a `CustomTransformSpec` changes in any way, the controller invalidates the cache entry for the `GroupResource` of the last known version of that spec. Additionally, if the change affected the `GroupResource` then the controller also invlidates the cache entry for the old `GroupResource`.
+
+The transport controller maintains a cached set of customization properties for each destination, and an association between `Binding` and the set of destinations that it references. When a relevant informer delivers an event about an inventory object (either a `ManagedCluster` object or a `ConfigMap` object that adds properties for that destination) the controller enqueues a work item of type `recollectProperties`. This work item carries the name of the inventory object. Processing that work item starts by re-computing the full map of properties for that destination. If the cache has an entry for that destination and the cached properties differ from the ones freshly computed, the controller updates that cache entry and enqueues a reference to every `Binding` object that depends on the properties of that destination.
```

Comment:
This paragraph touches back on parameter expansion after two unrelated paragraphs, and specifically after the one and only that mentions CustomTransforms. I understand the grouping since the one above is technical over caching, but it might make more sense to place this one after the paragraph that describes parameter expansion. I say this because I had to pause to understand that we now are back to a different customization feature.

Reply:
The paragraph about the caching for WEC-independent transforms precedes the paragraph about the caching for customizing to WECs because that is the order in which the two sorts of transformations are done and the order in which they are mentioned in the overall function paragraph.

We could swap the two paragraphs about caching, but with either order the second paragraph about caching is separated from the earlier words about what it is caching for.